openshift/templates/postgresql/postgresql-deploy.yaml

```diff
@@ -6,6 +6,36 @@ metadata:
     tags: ${NAME}${SUFFIX}
   name: ${NAME}${SUFFIX}-persistent-template
 objects:
+  - kind: NetworkPolicy
+    apiVersion: networking.k8s.io/v1
+    metadata:
+      name: ${NAME}
+      labels:
+        name: ${NAME}
+        app: ${APP_NAME}
+        env: ${TAG_NAME}
+    spec:
+      description: |
+        Allow the api(s) to access the db.
+      ingress:
+        - from:
+            - podSelector:
+                matchLabels:
+                  role: service
```

Comment:
The role of the service can be changed at deployment time by updating the parameters: this should also be populated by a parameter, in order to make it possible to update it if necessary.